w_common/lib/src/common/disposable.dart

```diff
@@ -514,7 +514,11 @@ class Disposable implements _Disposable, DisposableManagerV7, LeakFlagger {
 
   @mustCallSuper
   @override
-  T manageAndReturnTypedDisposable<T extends Disposable>(T disposable) {
+  T manageAndReturnTypedDisposable<T extends Disposable?>(T disposable) {
+    if (disposable == null) {
+      return disposable;
+    }
+
     _throwOnInvalidCall('manageAndReturnDisposable', 'disposable', disposable);
```

Comment:
It looks like we have a typo here. Maybe we should fix it while we're in here.

```suggestion
    _throwOnInvalidCall('manageAndReturnTypedDisposable', 'disposable', disposable);
```